frost/bip340.go

```diff
@@ -38,6 +38,40 @@ func (bc *Bip340Curve) EcBaseMul(k *big.Int) *Point {
 	return &Point{gs_x, gs_y}
 }
 
+// IsNotIdentity validates if the point lies on the curve and is not an identity
+// element.
+func (bc *Bip340Curve) IsNotIdentity(p *Point) bool {
+	return bc.IsOnCurve(p.X, p.Y)
+}
+
+// SerializedPointLength returns the byte length of a serialized curve point.
+func (b *Bip340Curve) SerializedPointLength() int {
+	// From the Marshal() function of secp256k1 go-ethereum implementation:
+	// 	 byteLen := (BitCurve.BitSize + 7) >> 3
+	//   ret := make([]byte, 1+2*byteLen)
+	return 65
+}
+
+// SerializePoint serializes the provided elliptic curve point to bytes.
+// The slice length is equal to SerializedPointLength().
+func (b *Bip340Curve) SerializePoint(p *Point) []byte {
+	// Note that secp256k1 implementation uses a fixed length of
+	// (BitCurve.BitSize + 7) >> 3
+	return b.Marshal(p.X, p.Y)
+}
+
+// DeserializePoint deserializes byte slice to an elliptic curve point. The
+// byte slice length must be equal to SerializedPointLength(). Otherwise,
+// the function returns nil.
```

Comment:
Should we also check that the deserialised points are valid points on the curve?

Reply:
Addressed in #6.